### DISCUSSION TITLE
Generating python parser - error

### DISCUSSION THREAD

**Discussion by alywonder** [tree-sitter/tree-sitter#1729] [Q&A]
I just started to experiment with Tree Sitter in a new project.  I am trying to setup parsers for different languages (Rust, Javascript, Typescript, Python, Go). I am using build.rs file. I found the following code excerpt to build the parser.

let mut dir: PathBuf = ["tree-sitter-rust", "src"].iter().collect();

 cc::Build::new()
        .include(&dir)
        .file(dir.join("parser.c"))
        .file(dir.join("scanner.c"))
        .compile("tree-sitter-rust");

It works for Rust, JS and TS. For Go, I did not see a scanner.c file, but it seems to work without using this file.
For python, I don't see scanner.c, but scanner.cc. I get link errors with or without using cc file. Is there anything I need to do differently here?

I am not sure if there is another forum to raise this issue. My apologies in advance if I am in the wrong place asking this question.

**Comment by ahelwer** [comment]
Don't grammar crates include their own build.rs file?

**Reply by alywonder** [reply]
Thank you.
Yes, they do. I was following the instructions given at
[Rust binding readme](https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_rust/README.md).
I also tried to bring the language specific repositories as submodules inside my project and compile them. I am getting linking errors. I am sure, I am doing something wrong!

**Reply by ahelwer** [reply]
Those docs are out of date unfortunately; I filed a bug about it at https://github.com/tree-sitter/tree-sitter/issues/1335

You can see a basic example of consuming a grammar from rust here: https://github.com/tlaplus-community/tlaplus-tool-dev-examples/tree/main/tree-sitter/rust

**Reply by alywonder** [reply]
Thank you very much for the response and the link. I was able to directly bring in different language repos and use it with Rust.
